src/pages/_app.tsx

```diff
@@ -61,12 +67,22 @@ const MyApp = ({ pageProps, Component }: AppProps) => {
         const id = new URLSearchParams(window.location.search)?.get('id');
         routerReplace(`/${id}`);
       },
+      isEmbbeded: pageProps.isEmbedded,
     }),
   );
   return (
     <AuthProvider value={authService}>
-      <Component {...pageProps} />
+      <EmbedProvider
+        value={{
+          ...embed,
+          isEmbedded: pageProps.isEmbedded,
+          originalUrl: withoutEmbedQueryParams(pageProps.query),
+        }}
+      >
```

Comment:
What's the reason for providing this at the root level? The auth service makes sense to be at the root, but the embed page might be better to be placed inside `App.tsx` - the idea being that we may end up having routes in the viz that don't care about the monaco editor or whether things are embedded or not.

Reply:
The reason I put it in _app was that your pages either are embedded or not but both need to pass on some metadata even when it's not the embedded page. So each page had to deal with passing that data manually where putting it in the _app will do it once for all pages.

Reply:
Why not in `App.tsx`, though? That is acting as a wrapper for everything we'd need embedded mode for - the Monaco editor, canvas etc.